pkg/kube/translation/apisix_route.go

```diff
@@ -133,6 +134,150 @@ func (t *translator) translateHTTPRouteNotStrictly(ctx *TranslateContext, ar *co
 	return nil
 }
 
+func (t *translator) TranslateRouteV2beta1(ar *configv2beta1.ApisixRoute) (*TranslateContext, error) {
+	ctx := &TranslateContext{
+		upstreamMap: make(map[string]struct{}),
+	}
+
+	if err := t.translateHTTPRouteV2beta1(ctx, ar); err != nil {
+		return nil, err
+	}
+	if err := t.translateStreamRoute(ctx, ar); err != nil {
+		return nil, err
+	}
+	return ctx, nil
+}
+
+func (t *translator) TranslateRouteV2beta1NotStrictly(ar *configv2beta1.ApisixRoute) (*TranslateContext, error) {
+	ctx := &TranslateContext{
+		upstreamMap: make(map[string]struct{}),
+	}
+
+	if err := t.translateHTTPRouteV2beta1NotStrictly(ctx, ar); err != nil {
+		return nil, err
+	}
+	if err := t.translateStreamRouteNotStrictly(ctx, ar); err != nil {
+		return nil, err
+	}
+	return ctx, nil
+}
+
+func (t *translator) translateHTTPRouteV2beta1(ctx *TranslateContext, ar *configv2beta1.ApisixRoute) error {
+	ruleNameMap := make(map[string]struct{})
+	for _, part := range ar.Spec.HTTP {
+		if _, ok := ruleNameMap[part.Name]; ok {
```

Comment:
Can we encapsulate this logic? It's so similar to the one in `v2alpha1`.

Reply:
I didn’t do this specially, because some fields were modified in the v2beta1 version in the [previous PR](https://github.com/apache/apisix-ingress-controller/pull/572), some pointers in v2alpha1 were removed.